### PR TITLE
add windows build support

### PR DESCRIPTION
**Pull request by AndrewDi**
### What problem does this PR solve? 
When building tidb on windows, go build retrun with some linux signal errors:
`# command-line-arguments
src\github.com\pingcap\tidb\tidb-server\main.go:441:34: undefined: syscall.SIGUSR1
src\github.com\pingcap\tidb\tidb-server\main.go:446:14: undefined: syscall.SIGUSR1 
` 

### What is changed and how it works?
add windows build support!
Change SIGUSR1 from go system control to user control by define custom signal:
`sigUsr1  := syscall.Signal(0x1e)`

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
1. Running tidb-server with MockStore
2. kill -30 $pid
3. Custom signal still can handle dump goroutine stack request.

Code changes

Side effects

Related changes
 - Need to be included in the release note
Support windows build, but build on windows should only be use in developer environment.

### CODE REVIEW

**Comment by shenli** [comment]
@AndrewDi Thanks!

**Comment by jackysp** [comment]
PTAL @lysu

**Comment by AndrewDi** [comment]
@lysu If I extract dump goroutine stack method to the other files, I think we'd better create new directory to place platform dependent code. Do you have suggest name?

**Comment by lysu** [comment]
how about put them into `util/signal` directory~? :laughing:

**Comment by AndrewDi** [comment]
> how about put them into `util/signal` directory~?

agree

File: util/signal/signal_windows.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+		syscall.SIGTERM,
+		syscall.SIGQUIT)
+
+	go func() {
+		sig := <-closeSignalChan
+		log.Infof("Got signal [%s] to exit.", sig)
+		if sig == syscall.SIGQUIT {
+			shudownFunc(true)
+		} else {
+			shudownFunc(false)
```
Comment (lysu) [inline util/signal/signal_windows.go]: maybe can simplify to `shutdownFunc(sig == syscall.SIGQUIT)`

File: util/signal/signal_windows.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import (
+	log "github.com/sirupsen/logrus"
+	"os"
+	"os/signal"
+	"syscall"
+)
+
+// SetupSignalHandler setup signal handler for TiDB Server
+func SetupSignalHandler(shudownFunc func(bool)) {
+
```
Comment (lysu) [inline util/signal/signal_windows.go]: maybe remove this line~

File: util/signal/signal_posix.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+		syscall.SIGTERM,
+		syscall.SIGQUIT)
+
+	go func() {
+		sig := <-closeSignalChan
+		log.Infof("Got signal [%s] to exit.", sig)
+		if sig == syscall.SIGQUIT {
+			shudownFunc(true)
+		} else {
+			shudownFunc(false)
```
Comment (lysu) [inline util/signal/signal_posix.go]: it's too

File: tidb-server/main.go [hunk 4]
```diff
@@ -45,14 +36,21 @@ import (
...
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/push"
 	log "github.com/sirupsen/logrus"
 	"google.golang.org/grpc"
+	"net"
+	"os"
+	"runtime"
+	"strconv"
+	"strings"
+	"time"
```
Comment (lysu) [inline tidb-server/main.go]: please move sys import to first import group....it's tidb's rule :laughing:

File: util/signal/signal_posix.go [hunk 5]
```diff
@@ -0,0 +1,57 @@
+// Copyright 2016 PingCAP, Inc.
```
Comment (lysu) [inline util/signal/signal_posix.go]: Copyright 2018 PingCAP

File: util/signal/signal_windows.go [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+// limitations under the License.
+// +build windows
+
+package signal
+
+import (
+	log "github.com/sirupsen/logrus"
+	"os"
+	"os/signal"
+	"syscall"
```
Comment (lysu) [inline util/signal/signal_windows.go]:
```
	"os"
 	"os/signal"
 	"syscall"

	log "github.com/sirupsen/logrus"
```

File: util/signal/signal_posix.go [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+// +build linux darwin freebsd unix
+
+package signal
+
+import (
+	log "github.com/sirupsen/logrus"
+	"os"
+	"os/signal"
+	"runtime"
+	"syscall"
```
Comment (lysu) [inline util/signal/signal_posix.go]:
```
 	"os"
 	"os/signal"
 	"runtime"
 	"syscall"

	log "github.com/sirupsen/logrus"
```

Comment (AndrewDi) [reply util/signal/signal_posix.go]: Thanks for your patient, I have do some modify.

**Comment by AndrewDi** [comment]
It looks good now...

**Pull request merged**